lib/ddtrace/buffer.rb

```diff
@@ -72,48 +126,103 @@ def close
         @closed = true
       end
     end
+  end
+
+  # Trace buffer that stores application traces and
+  # can be safely used concurrently with CRuby.
+  #
+  # Under extreme concurrency scenarios, this class can exceed
+  # its +max_size+ by up to 4%.
+  #
+  # Because singular +Array+ operations are thread-safe in CRuby,
```

Comment:
Is there a link to which Array operations are thread-safe in CRuby, or link to code? Might be something worth having in the comment to protect ourselves in the future

Reply:
Yes, it's in the `@see` section of this same documentation block, where it links to concurrent-ruby.